On branch feature/carol-assignment-branch

uxbycarol opened the pull request:
push test

uxbycarol committed:
On branch feature/carol-assignment-branch
  push test

uxbycarol committed:
Changes to be committed:
    modified:   src/routes/+page.svelte
      layout the basic structure of the section
      write the copy for the section
	new file:   static/cost-management-image.png

uxbycarol committed:
Changes to be committed:
modified:   src/app.css
    adding variables for colors
    adding styles for buttons
modified:   src/routes/+page.svelte
    adding color styles

uxbycarol committed:
Changes to be committed:
	modified:   src/routes/+page.svelte
       added background gradient
       polished the layout
       polished some styles

uxbycarol committed:
Changes to be committed:
	modified:   src/routes/+page.svelte

uxbycarol committed:
Changes to be committed:
	modified:   src/routes/+page.svelte
        from HTML to svelte component
	new file:   src/routes/feature.svelte

uxbycarol commented:
Hi buhodev,  I would like to include the check icon in mi design as in the image. While I could add the SVG manually, I prefer to integrate it as a icon component.

Could you please provide me with the code for the icon component?.

Thank you for your assistance.

![image](https://github.com/user-attachments/assets/2a9fd9e0-db88-4def-abef-e0c97d180fdc)

buhodev commented:
We're not using a component for the icons, we're using a preprocessor that creates the html at build time (https://github.com/unplugin/unplugin-icons). But a simple component for the Check icon will be this:

```svelte
<script lang="ts">
	let className: string | undefined = 'size-5'
	export { className as class };
</script>

<!-- SVG of the "Check" Mini icon copied from https://heroicons.com -->
<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 20 20" fill="currentColor" class={className}>
	<path
		fill-rule="evenodd"
		d="M16.704 4.153a.75.75 0 0 1 .143 1.052l-8 10.5a.75.75 0 0 1-1.127.075l-4.5-4.5a.75.75 0 0 1 1.06-1.06l3.894 3.893 7.48-9.817a.75.75 0 0 1 1.05-.143Z"
		clip-rule="evenodd"
	/>
</svg>
```

I'd recommend to just use the svg icon and modify the properties in the html rather than use a component

uxbycarol committed:
Changes to be committed:
	modified:   src/routes/+page.svelte
     added the icons to the especifications of the feature
	new file:   src/routes/checkIcon.svelte
        Finished the design

uxbycarol committed:
Changes to be committed:
    Change the checkIcon component for just svg in the HTML

uxbycarol commented:
Thanks, I followed your recommendation. The assignment is ready for review

buhodev commented on src/app.css:
```diff
@@ -2,6 +2,51 @@
+    --on-warning: #000000;
+    --on-error: #000000;
+    --on-surface: #ffffff;
+    --color-primary-50: #fdf1e9;
```
Tailwind already generates all the utilities based on the colors that you added from line 18 until line 47. They're defined in the `tailwind.config.js` file. If you're not using the css variables somewhere else, I'd remove them from here

buhodev commented on src/routes/+page.svelte:
```diff
@@ -1,3 +1,88 @@
+		ctaLink="#"
+		imageSrc="/cost-management-image.png"
+		imageAlt="Cost Management Feature"
+		><slot>
```
It's not necessary to wrap the markup inside of a `<slot>` tag here. You can safely remove the `slot`

uxbycarol replied:
Oh, thanks

buhodev commented on src/routes/feature.svelte:
```diff
@@ -0,0 +1,25 @@
+        <div class="flex flex-col basis-3/5 gap-3 md:gap-6">
+            <h2 class="text-2xl xl:text-4xl font-bold text-white">{title}</h2>
+            <p class="text-pretty text-base xl:text-xl text-gray-300">{description}</p>
+          <slot></slot>
```
Here it's ok to keep the `<slot>` tag

buhodev reviewed (changes requested):
Good job, Carol. Please run `bun run format` inside of your terminal to format the files after addressing the comments I left bellow, and commit the changes

uxbycarol committed:
Changes to be committed:
   Applied feedback from Carlos:
        Eliminated the variables from app.css.
        Removed the slot markup from +page.svelte.
        Formatted the files
